libs/csm/rest/csm_rest_query_deployment.py

```diff
@@ -75,47 +96,39 @@ def get_system_topology(self, uri_param=None, auth_header=None):
         return response
 
     # Function not ready
-    def verify_system_topology(self, deploy_version, expected_response=HTTPStatus.OK):
+    def verify_system_topology(self, deploy_start_time: str, deploy_end_time: str,
+                             expected_response=HTTPStatus.OK):
         """
         Verify Get system topology
         """
-        err_msg = ""
-        unique_list = []
-        time_list= []
-        resp = self.get_system_topology()
-        result = True
-        if resp.status_code == expected_response:
-            result = True
-            get_response = resp.json()
-            self.log.info("1. Verify id fields have unique values")
-            for elements in get_response["topology"]:
-                if 'id' in elements.keys():
-                    unique_list.append(elements['id'])
-            if len(set(unique_list)) == len(unique_list):
-                self.log.info("id fields have unique values")
-            else:
-                err_msg = "Duplicate values found for id fields"
-                self.log.error(err_msg)
-                result = False
-            self.log.info("2. Verify deployment version and time")
-            response_version = get_response["topology"]["version"]
-            if response_version == deploy_version:
-                self.log.info("Version match found")
-            else:
-                err_msg = err_msg + "Version mismatch found"
-                self.log.error(err_msg)
-                result = False
-            for nodes in get_response["topology"]["nodes"][0]:
-                time_list.append(nodes["deployment_time"])
-            #compare deployment times
-            #verify node details(call verify storage details inside)
-            self.log.info("Verify certificate details")
+        get_response = self.get_system_topology()
+        self.log.info("Get system topology response %s", get_response)
+        result = get_response.status_code == expected_response
+        if result:
+            self.log.info("Step 1: Verify id fields have unique values")
+            result, err_msg = self.verify_unique_ids()
+            assert result, err_msg
+            self.log.info("Step 2: Verify deployment version and time")
+            result, err_msg = self.verify_deployment_version_time(get_response,
+                             deploy_start_time, deploy_end_time)
+            assert result, err_msg
+            self.log.info("Step 3: Verify node details")
+            result, err_msg = self.verify_node_details(get_response)
+            assert result, err_msg
+            self.log.info("Step 4: Verify storage details")
+            result, err_msg = self.verify_storage_details(get_response)
+            assert result, err_msg
```

Comment:
can we append the err_msg, and not assert multiple times ?

Reply:
as per previous discussion, error messages would be appended in individual functions and assert can be used in main function since main function mostly consists of test steps

Reply:
ok